src/utils/context.rs

```diff
@@ -1,23 +1,32 @@
+use std::sync::Arc;
+
+use rocket::request::{self, FromRequest, Request};
+
 use crate::cache::Cache;
 use crate::config::scheme;
 use crate::utils::http_client::HttpClient;
-use rocket::request::{self, FromRequest, Request};
-use std::sync::Arc;
+use crate::ServiceCache;
+
+pub(crate) const UNSPECIFIED_CHAIN: &'static str = "UNSPECIFIED";
 
 pub struct RequestContext {
     pub request_id: String,
     pub host: String,
     http_client: Arc<dyn HttpClient>,
     cache: Arc<dyn Cache>,
+    mainnet_cache: Arc<ServiceCache>,
```

Comment:
Why is this a `ServiceCache` and the other is a `dyn Cache`?

Reply:
I'll have a look into this one. The state managed by Rocket cannot have the same type twice (eg.: two `dyn Cache`) but for the `RequestContext` it shouldn't be a problem having what you mentioned.